### PR TITLE
Repair parser tests for new fmt, fix leaver edge case

### PR DESCRIPTION
**Pull request by prsauer**

### CODE REVIEW

File: packages/parser/src/pipeline/retail/segmentToCombat.ts [hunk 1]
```diff
@@ -211,6 +211,12 @@ export const segmentToCombat = () => {
             recentShuffleRoundsBuffer = [];
             recentScoreboardBuffer = [];
             return shuf;
+          } else {
+            // We hit a final round (ARENA_MATCH_END) but the Match itself wasn't a valid 6-round shuffle
+            // We want to emit the shuffle as a round but then reset the internal match aggregator
+            recentShuffleRoundsBuffer = [];
+            recentScoreboardBuffer = [];
+            return decoded.shuffle;
```
Comment (kunchenguid) [inline packages/parser/src/pipeline/retail/segmentToCombat.ts:219]: what's in this shuffle here?

File: packages/parser/test/shuffle_early_leaver.test.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  beforeAll(() => {
+    const loaded = loadLogFile('shuffle_early_leaver.txt');
+    results.combats = loaded.combats;
+    results.malformedCombats = loaded.malformedCombats;
+    results.shuffleRounds = loaded.shuffleRounds;
+    results.shuffles = loaded.shuffles;
+  });
+
+  it('should return a single match', () => {
```
Comment (kunchenguid) [inline packages/parser/test/shuffle_early_leaver.test.ts:21]: this says single match but seems we're expecting 2 rounds and 0 shuffles. a little confused.. :)

**Review by kunchenguid — APPROVED** [review]
